### ISSUE TITLE
AVX2 issue

### ISSUE THREAD

**Issue by jepminer** [deeppomf/DeepCreamPy#19]
'Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2'
I just get this when trying to run the latest windows binary.
![image](https://user-images.githubusercontent.com/33632057/47575981-48a6f800-d93b-11e8-8602-d7719a6d1830.png)

**Comment by anounyym1** [comment]
I am having same issue. Using i7-4790K CPU so AVX2 should be supported.

**Comment by deepWhomst** [comment]
That indicates there are CPU extensions that are not being taken advantage of to increase the tensorflow binary's processing speed. You don't really need to worry about that, they're left out to ensure a higher compatibility with CPUs that lack features, since there isn't really a way to do CPUID feature flag detection with tensorflow. 
In short, it's just notifying you that in order to make the program available to everyone, some less common cpu features are not being used.

**Comment by jepminer** [comment]
> That indicates there are CPU extensions that are not being taken advantage of to increase the tensorflow binary's processing speed. You don't really need to worry about that, they're left out to ensure a higher compatibility with CPUs that lack features, since there isn't really a way to do CPUID feature flag detection with tensorflow.
> In short, it's just notifying you that in order to make the program available to everyone, some less common cpu features are not being used.

but it's not just notifying me, it's stopping the program from working. It doesn't or can't progress beyond this message.

**Comment by deepWhomst** [comment]
Try python -v decensor.py

**Comment by jepminer** [comment]
> Try python -v decensor.py

assuming I have to do it the other way and not using the pre-done download (mainly because I tried and nothing worked) I was using when I try and gather the dependencies I get this:
![image](https://user-images.githubusercontent.com/33632057/47596249-da832500-d97c-11e8-8f31-110b0dd7dc87.png)

**Comment by deeppomf** [comment]
Are you using Python 2? Tensorflow is not compatible with Python 2 on Windows.

**Comment by jepminer** [comment]
> Are you using Python 2? Tensorflow is not compatible with Python 2 on Windows.

python 3

**Comment by deeppomf** [comment]
Try updating your pip with 
`python -m pip install --upgrade pip`
Is your Python 64-bits? Tensorflow is not compatible with 32 bit Python.

**Comment by jepminer** [comment]
> Try updating your pip with
> `python -m pip install --upgrade pip`
> Is your Python 64-bits? Tensorflow is not compatible with 32 bit Python.

I'd already upgraded pip and it turns out it wasn't but I'm using 3.7.1 64bit now and getting the exact same error.

**Comment by deeppomf** [comment]
I got it! Python 3.7 is not compatible with Tensorflow yet.
https://github.com/tensorflow/tensorflow/issues/20517

**Comment by jepminer** [comment]
> I got it! Python 3.7 is not compatible with Tensorflow yet.
> [tensorflow/tensorflow#20517](https://github.com/tensorflow/tensorflow/issues/20517)

it started doing a whole bunch of stuff finally but still errored out complaining about AVX2 and then just cleaned everything up and stopped

**Comment by deepWhomst** [comment]
Pleas post the log of exactly what it said. If possible use python -v uncensor.py

**Comment by jepminer** [comment]
> How did you go about resolving the Python 3.7 incompatibility? Did you uninstall python 3.7 and install python 3.6? Did you create a new environment in anaconda with 3.6? etc. Please describe your process with as much detail as you can.
> Also, please post logs of exactly what it said. If possible use `python -v uncensor.py`

I just uninstalled 3.7 and installed the latest 3.6 and tried again nothing fancy.
![image](https://user-images.githubusercontent.com/33632057/47609298-db2ac280-da33-11e8-8028-f542521e1c7e.png) I was able to get this image of the end of everything that happens before it cleans up but this is all I can get because for some reason doing a redirect into a txt file isn't working for any python command, when I tested it with just a DIR command it works fine.

**Comment by deeppomf** [comment]
Have you tried installing a tensorflow version that supports AVX from https://github.com/fo40225/tensorflow-windows-wheel?

**Comment by segop** [comment]
I also have the AVX2 issue.
I installed the latest python 3.6 and tensorflow 0.11.0
Still get the issue, but would tensorflow 0.11.0 be the right one or should i try another version?

**Comment by deeppomf** [comment]
What about doing a clean Python install and following the steps in https://github.com/deeppomf/DeepCreamPy/issues/26#issuecomment-434043166?

**Comment by OliverFrWTFalreadyTakenComeOnNow** [comment]
> 
> 
> I also have the AVX2 issue.
> I installed the latest python 3.6 and tensorflow 1.11.0
> Still get the issue, but would tensorflow 1.11.0 be the right one or should i try another version?
> 
> EDIT: I uninstalled tensorflow and installed "1.11.0/py36/CPU/avx2/tensorflow-1.11.0-cp36-cp36m-win_amd64.whl" from the link you gave. I don't get the AVX2 warning anymore but there isn't an output.
> This is all that happens when i run "python decensor.py":
> ![capture](https://user-images.githubusercontent.com/44552371/47682113-cbc98780-dba1-11e8-989a-bfc5393feecc.PNG)

Just in case: are you sure you fed the program a PNG image, and not a .jpg? Don't laugh, because I didn't notice that detail, I lost an entire afternoon's spare available hours yesterday, so maybe I'm not alone in this ;)

**Comment by pmp45** [comment]
I have the same issue with decensor.exe. I hope for a fix someday.

**Comment by segop** [comment]
> Just in case: are you sure you fed the program a PNG image, and not a .jpg? Don't laugh, because I didn't notice that detail, I lost an entire afternoon's spare available hours yesterday, so maybe I'm not alone in this ;)

Hah, I thought I had read all of the instructions, the input I had was indeed a .jpg. Thanks a lot for the help.

**Comment by anounyym1** [comment]
> Hah, I thought I had read all of the instructions, the input I had was indeed a .jpg. Thanks a lot for the help.
Might there should be "No PNG-files found on decensor_input-folder, exiting"-warning?

**Comment by Miyazono1** [comment]
so i've done every step, installed tensorflow, and all that, but i'm still getting this issue. how can we tell what version of tensorflow is the right one that won't produce this error? and do i need to uninstall my current version before installing a different one?

**Comment by deeppomf** [comment]
What did you do to get it working?
618 censored regions is too high. Could you share the test image with me via reddit or twitter?
It should be python -v decensor.py --is_mosaic=true
Try doing a bar decensor. Sometimes the neural network doesn't detect the mosaic region and does nothing even though you correctly set up the images.

**Comment by jepminer** [comment]
I downloaded and installed this file https://github.com/fo40225/tensorflow-windows-wheel/tree/master/1.11.0/py36/CPU/sse2 and placed it into my downloaded master branch copy, pip installed it and yeah I must have messed up big time, tried it again remaking the image from scratch and it detected all 3 regions and worked but boy were you not kidding when you said mosaic wasn't the best. It's working now though so that's good

**Comment by Kadantte** [comment]
@jepminer That's what i did to and it's working fine, @deeppomf that's why i said you should add that to readme for people who has the same problem.

**Comment by deeppomf** [comment]
@Kadantte I did link https://github.com/deeppomf/DeepCreamPy/issues/26 in the troubleshooting part of the README, but I'll make it more explicit.
